File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/authorization/AllowAllSemanticAuthorizationPolicy.java
package com.foundgine.core.semantic.authorization;
import com.foundgine.core.abstractions.*;
/** Default unrestricted policy. */
public class AllowAllSemanticAuthorizationPolicy implements ISemanticAuthorizationPolicy {
 public boolean canAccessEntity(EntityId id){return true;} public boolean canAccessField(EntityId e,FieldId f){return true;} public boolean canAccessRelationship(EntityId e,RelationshipId r){return true;}
 public boolean canWriteEntity(EntityId id){return true;} public boolean canWriteField(EntityId e,FieldId f){return true;} public boolean canWriteRelationship(EntityId e,RelationshipId r){return true;}
 public AuthorizationPredicate getPredicate(EntityId e,AuthorizationOperation o){return null;}
 public AuthorizationDecision getEntityAccess(EntityId e,AuthorizationOperation o){return o==AuthorizationOperation.READ?(canAccessEntity(e)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED):(canWriteEntity(e)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED);}
 public AuthorizationDecision getEntityAccess(EntityId e,AuthorizationOperation o,AuthorizationOperationName n){return getEntityAccess(e,o);}
 public AuthorizationDecision getFieldAccess(EntityId e,FieldId f,AuthorizationOperation o){return o==AuthorizationOperation.READ?(canAccessField(e,f)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED):(canWriteField(e,f)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED);}
 public AuthorizationDecision getRelationshipAccess(EntityId e,RelationshipId r,AuthorizationOperation o){return o==AuthorizationOperation.READ?(canAccessRelationship(e,r)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED):(canWriteRelationship(e,r)?AuthorizationDecision.ALLOWED:AuthorizationDecision.DENIED);}
}

File: src/csharp/Foundgine.Core/Semantic/Authorization/AllowAllSemanticAuthorizationPolicy.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Authorization;

/// <summary>Default unrestricted policy for applications that do not need authorization.</summary>
public class AllowAllSemanticAuthorizationPolicy : ISemanticAuthorizationPolicy
{
    public virtual bool CanAccessEntity(EntityId entityId) => true;
    public virtual bool CanAccessField(EntityId entityId, FieldId fieldId) => true;
    public virtual bool CanAccessRelationship(EntityId sourceEntityId, RelationshipId relationshipId) => true;
    public virtual bool CanWriteEntity(EntityId entityId) => true;
    public virtual bool CanWriteField(EntityId entityId, FieldId fieldId) => true;
    public virtual bool CanWriteRelationship(EntityId sourceEntityId, RelationshipId relationshipId) => true;

    public virtual AuthorizationPredicate? GetPredicate(
        EntityId entityId,
        AuthorizationOperation operation) => null;

    public virtual AuthorizationDecision GetEntityAccess(
        EntityId entityId,
        AuthorizationOperation operation) =>
        operation == AuthorizationOperation.Read
            ? (CanAccessEntity(entityId) ? AuthorizationDecision.Allowed : AuthorizationDecision.Denied)
            : (CanWriteEntity(entityId) ? AuthorizationDecision.Allowed : AuthorizationDecision.Denied);

    /// <summary>
    /// Named-operation refinement of the coarse Read/Write decision above.
    /// Declared as a real virtual class member (not just an interface default
    /// method) so derived policies can use <c>override</c> to layer
    /// additional, domain-specific requirements onto specific operation
    /// names. The default simply falls back to the coarse decision, matching
    /// <see cref="ISemanticAuthorizationPolicy"/>'s own default.
    /// </summary>
    public virtual AuthorizationDecision GetEntityAccess(
        EntityId entityId,
        AuthorizationOperation operation,
        AuthorizationOperationName? name) =>
        GetEntityAccess(entityId, operation);

    public virtual AuthorizationDecision GetFieldAccess(
        EntityId entityId,
        FieldId fieldId,
        AuthorizationOperation operation) =>
        operation == AuthorizationOperation.Read
            ? (CanAccessField(entityId, fieldId) ? AuthorizationDecision.Allowed : AuthorizationDecision.Denied)
            : (CanWriteField(entityId, fieldId) ? AuthorizationDecision.Allowed : AuthorizationDecision.Denied);

    public virtual AuthorizationDecision GetRelationshipAccess(
        EntityId sourceEntityId,
        RelationshipId relationshipId,
        AuthorizationOperation operation) =>
        operation == AuthorizationOperation.Read
            ? (CanAccessRelationship(sourceEntityId, relationshipId)
                ? AuthorizationDecision.Allowed
                : AuthorizationDecision.Denied)
            : (CanWriteRelationship(sourceEntityId, relationshipId)
                ? AuthorizationDecision.Allowed
                : AuthorizationDecision.Denied);
}